test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java

```diff
@@ -265,6 +275,423 @@ public void testGetDistanceBetweenTwoSCwithStartSCAsUnboundedAndStreamTruncateda
         assertEquals(Long.valueOf(20), distance.join());
     }
 
+    @Test(timeout = 50000)
+    public void testNextStreamCut() throws ExecutionException, InterruptedException {
+        StreamConfiguration config = StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.fixed(1))
+                .build();
+        Controller controller = controllerWrapper.getController();
+        controllerWrapper.getControllerService().createScope(SCOPE + "-1", 0L).get();
+        controller.createStream(SCOPE + "-1", STREAM + "-1", config).get();
+        @Cleanup
+        ConnectionFactory connectionFactory = new SocketConnectionFactoryImpl(ClientConfig.builder().build());
+        @Cleanup
+        ClientFactoryImpl clientFactory = new ClientFactoryImpl(SCOPE + "-1", controller, connectionFactory);
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(STREAM + "-1", new JavaSerializer<>(),
+                EventWriterConfig.builder().build());
+        // write events to stream
+        write30ByteEvents(10, writer);
+        Segment segment = new Segment(SCOPE + "-1", STREAM + "-1", 0);
+        StreamCut streamCut = new StreamCutImpl(Stream.of(SCOPE + "-1", STREAM + "-1"),
```

Comment:
Add scenario for 2 segments in a stream rather than having single segment.

Reply:
This is simple flow with one segment. However, multiple segments flow have been tested as part of other test cases.